app-creator/src/redux/types/attributes.ts

```diff
@@ -0,0 +1,283 @@
+/**
+ *  @fileoverview This file contains interface types for the different style attributes.
+ */
+import { InputType } from './enums';
+
+export type SharedAttributes =
+  | 'height'
+  | 'width'
+  | 'padding'
+  | 'margin'
+  | 'color'
+  | 'backgroundColor'
+  | 'borderWidth'
+  | 'borderType'
+  | 'borderColor'
+  | 'fontSize'
+  | 'fontWeight'
+  | 'fontFamily'
+  | 'textAlign'
+  | 'whiteSpace'
+  | 'shown';
+
+export interface AttributeMetaData {
+  [key: string]: {
+    value: string;
+    type: InputType;
+    items?: string[];
+  };
+}
+
+export interface DefaultAttributesType {
+  [key: string]: string;
+}
+
+export const sharedAttributes: AttributeMetaData = {
+  height: {
+    value: '',
+    type: InputType.number,
+  },
+  width: {
+    value: '',
+    type: InputType.number,
+  },
+  padding: {
+    value: '0',
+    type: InputType.number,
+  },
+  margin: {
+    value: '8',
+    type: InputType.number,
+  },
+  color: {
+    value: 'black',
+    type: InputType.color,
+  },
+  backgroundColor: {
+    value: 'none',
+    type: InputType.color,
+  },
+  borderWidth: {
+    value: '0',
+    type: InputType.number,
+  },
+  borderType: {
+    value: 'solid',
+    type: InputType.select,
+    items: ['solid', 'dashed'],
+  },
+  borderColor: {
+    value: 'black',
+    type: InputType.color,
+  },
+  fontSize: {
+    value: '12',
+    type: InputType.number,
+  },
+  fontWeight: {
+    value: '300',
+    type: InputType.select,
+    items: ['100', '200', '300', '400', '500', '600', '700', '800', '900'],
+  },
+  fontFamily: {
+    value: 'Roboto',
+    type: InputType.select,
+    items: [
+      'Arial',
+      'Arial Black',
+      'Bookman',
+      'Candara',
+      'Comic Sans MS',
+      'Courier',
+      'Courier New',
+      'Garamond',
+      'Georgia',
+      'Impact',
+      'Open Sans',
+      'Palatino',
+      'Roboto',
+      'Times',
+      'Times New Roman',
+      'Verdana',
+    ],
+  },
+  textAlign: {
+    value: 'left',
+    type: InputType.select,
+    items: ['left', 'center', 'right', 'justify'],
+  },
+  whiteSpace: {
+    value: 'normal',
+    type: InputType.select,
+    items: ['normal', 'pre'],
+  },
+  shown: {
+    value: 'true',
+    type: InputType.select,
+    items: ['true', 'false'],
+  },
+};
+
+export const DEFAULT_SHARED_ATTRIBUTES: DefaultAttributesType = Object.keys(
+  sharedAttributes
+).reduce((attributes, key) => {
+  if (sharedAttributes[key].type === InputType.number) {
+    return { ...attributes, [key]: sharedAttributes[key].value + 'px' };
+  }
+  return { ...attributes, [key]: sharedAttributes[key].value };
+}, {});
+
+export const textboxAttributes: AttributeMetaData = {
```

Comment:
I was wondering how you felt about making these accessible as a static "attributes" property on each of the individual widget components? I feel like it's cleaner to have the widget specific attributes defined locally to each widget, and anything shared attributes or interfaces can be defined here?